draft-ietf-masque-h3-datagram.md

```diff
@@ -180,56 +161,57 @@ of {{QUIC}}):
 
 ~~~
 HTTP/3 Datagram {
-  Quarter Stream ID (i),
+  Encoded Stream ID (i),
   [Context ID (i)],
   HTTP Datagram Payload (..),
 }
 ~~~
 {: #h3-datagram-format title="HTTP/3 DATAGRAM Format"}
 
-Quarter Stream ID:
-
-: A variable-length integer that contains the value of the client-initiated
-bidirectional stream that this datagram is associated with, divided by four (the
-division by four stems from the fact that HTTP requests are sent on
-client-initiated bidirectional streams, and those have stream IDs that are
-divisible by four). The largest legal QUIC stream ID value is 2<sup>62</sup>-1,
-so the largest legal value of Quarter Stream ID is 2<sup>62</sup>-1 / 4.
-Receipt of a frame that includes a larger value MUST be treated as a connection
-error of type FRAME_ENCODING_ERROR.
+Encoded Stream ID:
+
+: A variable-length integer that contains an encoded value indicating the
+client-initiated bidirectional stream that this datagram is associated with.
+The Encoded Stream ID for a datagram without a context ID is the associated
+stream ID divided by two. The Encoded Stream ID for a datagram with a context
+ID is the associated stream ID divided by two, plus one. (Client-initiated
+bidirectional streams have stream IDs that are divisible by four, so the
+stream ID divided by two always has the least-significant bit set to zero).
+The largest legal QUIC stream ID value is 2<sup>62</sup>-1, so the largest legal
+value of Encoded Stream ID is (2<sup>62</sup>-1 / 2) + 1. Receipt of a frame that
+includes a larger value MUST be treated as a connection error of type
+FRAME_ENCODING_ERROR. A datagram without a context ID is implicitly using
+a context ID of 0.
 
 Context ID:
 
 : A variable-length integer indicating the context ID of the datagram (see
 {{datagram-contexts}}). Whether or not this field is present depends on whether
-datagram contexts are in use on this stream, see {{context-hdr}}. If this QUIC
-DATAGRAM frame is reordered and arrives before the receiver knows whether
-datagram contexts are in use on this stream, then the receiver cannot parse
-this datagram and the receiver MUST either drop that datagram silently or
-buffer it temporarily.
+the least-significant bit of the Encoded Stream ID, as described above. If the
+receiver of a datagram does not support Context IDs, it will drop any datagram
+that contains a context ID.
```

Comment:
Should we say something like: `If the Context ID field is not present, this datagram is associated with the default context (which has its Context ID set to 0).`

Reply:
```suggestion
that contains a context ID. If the Context ID field is not present, this datagram
is associated with the default context (which has its Context ID set to 0).
```